Lib/sqlite3/test/backup.py

```diff
@@ -0,0 +1,166 @@
+            self.cx.backup(bckfn, pages=1, progress='bar')
+        self.assertEqual(str(err.exception), 'progress argument must be a callable')
+
+    def test_modifying_progress(self):
+        journal = []
+
+        def progress(status, remaining, total):
+            if not journal:
+                self.cx.execute('INSERT INTO foo (key) VALUES (?)', (remaining+1000,))
+                self.cx.commit()
+            journal.append(remaining)
+
+        bckfn = self.temp_file_name
+        self.cx.backup(bckfn, pages=1, progress=progress)
+        self.verify_backup(bckfn)
+
+        cx = sqlite.connect(bckfn)
+        result = cx.execute("SELECT key FROM foo"
+                            " WHERE key >= 1000"
+                            " ORDER BY key").fetchall()
+        self.assertEqual(result[0][0], 1001)
+
+        self.assertEqual(len(journal), 3)
+        self.assertEqual(journal[0], 1)
+        self.assertEqual(journal[1], 1)
+        self.assertEqual(journal[2], 0)
+
+    def test_failing_progress(self):
+        def progress(status, remaining, total):
+            raise SystemError('nearly out of space')
+
+        bckfn = self.temp_file_name
+        with self.assertRaises(SystemError) as err:
+            self.cx.backup(bckfn, progress=progress)
+        self.assertEqual(str(err.exception), 'nearly out of space')
+        self.assertFalse(os.path.exists(bckfn))
+
+    def test_database_source_name(self):
+        bckfn = self.temp_file_name
+        self.cx.backup(bckfn, name='main')
+        self.cx.backup(bckfn, name='temp')
+        with self.assertRaises(sqlite.OperationalError):
+            self.cx.backup(bckfn, name='non-existing')
+        self.assertFalse(os.path.exists(bckfn))
+        self.cx.execute("ATTACH DATABASE ':memory:' AS attached_db")
+        self.cx.execute('CREATE TABLE attached_db.foo (key INTEGER)')
+        self.cx.executemany('INSERT INTO attached_db.foo (key) VALUES (?)', [(3,), (4,)])
+        self.cx.commit()
+        bckfn = self.temp_file_name
+        self.cx.backup(bckfn, name='attached_db')
+        self.verify_backup(bckfn)
+
+    def test_backup_to_other_connection(self):
+        dx = sqlite.connect(':memory:')
+        self.cx.backup(dx)
+        result = dx.execute("SELECT key FROM foo ORDER BY key").fetchall()
+        self.assertEqual(result[0][0], 3)
+        self.assertEqual(result[1][0], 4)
+
+    def test_backup_to_other_connection_in_transaction(self):
+        dx = sqlite.connect(':memory:')
+        dx.execute('CREATE TABLE bar (key INTEGER)')
+        dx.executemany('INSERT INTO bar (key) VALUES (?)', [(3,), (4,)])
+        try:
+            with self.assertRaises(sqlite.OperationalError):
+                self.cx.backup(dx)
+        finally:
+            dx.rollback()
+
+def suite():
+    return unittest.TestSuite(unittest.makeSuite(BackupTests, "test_"))
+
+def test():
+    runner = unittest.TextTestRunner()
+    runner.run(suite())
+
+if __name__ == "__main__":
+    test()
```

Comment:
Does not `unittest.main()` work?

Reply:
Yes, now that the backup suite uses _snake_like_ method names. Done.